chain-impl-mockchain/tests/common/address.rs

```diff
@@ -24,6 +40,19 @@ impl AddressData {
         }
     }
 
+    pub fn as_input(&self, value: Value, utxo: UtxoPointer) -> Input {
+        match self.address.kind() {
+            Kind::Account { .. } => {
+                Input::from_account_public_key(self.public_key.clone(), value.clone())
+            }
+            _ => Input::from_utxo(utxo),
```

Comment:
This looks dodgy. The match should be exhaustive, so that adding a new address variant would make you revisit the implementation here. All the other kinds can be listed in the match arm separated with ` | `.